### ISSUE TITLE
Cannot create Take Profit Limit order

### ISSUE THREAD

**Issue by CoenraadS** [JKorf/Kraken.Net#44]
**Describe the bug**
Take Profit Limit order claims not enough available balance

**To Reproduce**
```
var ticker = await api.GetTickersAsync(BTCEUR)();
            var btcAskPrice = ticker.Data[BTCEUR].BestAsks.Price;

            var balances = await api.GetAvailableBalancesAsync()();
            var euros = GetBalance(balances.Data, EURO);
            var volume = euros / btcAskPrice;

            var percent = 1.5M;

            var sellPrice = Math.Round(btcAskPrice * (1 + (percent / 100)), 1);

           var order = await api.PlaceOrderAsync(
                BTCEUR, OrderSide.Buy,
                OrderType.TakeProfitLimit,
                volume,
                price: btcAskPrice,
                secondaryPrice: sellPrice,
                validateOnly: dryRun
                )();

        private static decimal GetBalance(Dictionary<string, KrakenBalanceAvailable> balances, string currency)
        {
            if (balances.TryGetValue(currency, out var balance))
            {
                return balance.Available;
            }

            return 0;
        }
```

**Expected behavior**
Order should be placed. When using a `TakeProfit` order with the same code, it is successful.

**Comment by JKorf** [comment]
Have a look at the logging. If you set the LogLevel/LogVerbosity to Debug in the client options it will output the sent and received data.

**Comment by EpsilonD3lta** [comment]
I think the problem is in your code. You calculate volume with "btcAskPrice" and then try to buy it with (oddly named) "sellPrice" which is higher by 1.5 percent, and therefore "not enough balance" can happen. Either you have actually want to have OrderSide.Sell or you want to buy with lower price or I don't understand your code at all.

**Comment by CoenraadS** [comment]
Hmm, it could be I completly misread how this api is supposed to work. I thought I could place an order, and when it fills automatically a sell order would be placed at 1.5%+ the filled order.

**Comment by EpsilonD3lta** [comment]
No that is not how take profit limit works on Kraken: https://support.kraken.com/hc/en-us/articles/360029116112-Take-profit-limit-orders
TLDR: one price is for triggering (no order is created in order book until price reaches trigger price - Kraken exchange just watches the prices for you), other price is a limit price for the actual order, that Kraken exchange creates for you when trigger price is reached.

What you probably want to do (according to your last comment) is a conditional close order, which Kraken supports, but I think that this library does not yet support this feature.

**Comment by CoenraadS** [comment]
Indeed: https://support.kraken.com/hc/en-us/articles/360038640052-Conditional-Close

It's is a conditional close of type 'Take Profit'
